### DISCUSSION TITLE
Get Markdown source code from posts which you cannot edit

### DISCUSSION THREAD

**Discussion by porg** [community/community#17667] [Projects and Issues]
## User Story

![Github - Get Markdown source code from posts which you cannot edit 1](https://user-images.githubusercontent.com/737143/171621311-6455d911-8422-4ad2-b569-748ce94d9912.png)

![Github - Get Markdown source code from posts which you cannot edit 2](https://user-images.githubusercontent.com/737143/171621305-396b9219-00f2-41bc-8f93-39441091e2f5.png)

![Github - Get Markdown source code from posts which you cannot edit 3](https://user-images.githubusercontent.com/737143/171623451-06860357-b8cb-4763-bfc9-2b9d985222a1.png)

## Problems
1. Some markdown elements are broken in their quoted representation.
a) E.g. the table markup is not a table markup anymore.
2. Even if everything gets quoted in order, I need to manually remove the quoting character `>` from the beginning of each line, which I want to reuse for my altered content.

## Proposals
- Ad 1a) should be fixed. It should be possible to reliably quote a table.
- Ad 2) Add the action "Copy post source code" (Github flavored Markdown) to the `…` dropdown context menus of the initial post and all the comments in an issue thread.

**Comment by porg** [comment]
No-one troubled by the problem?
No-one interested in the solution / improvement?

**Comment by melvincwng** [comment]
@porg, thanks for bringing up this issue to the community.

Just FYI, in order to get the markdown source code, an alternative way is to simply just copy the original table from the post. Basically just do a Cltrl C and Cltrl V. That works as well.

Regarding your proposed method of 'Quote reply', it could be a potential bug. Hence, I would kindly suggest you to raise a ticket to [GitHub Bug Report](https://support.github.com/contact/bug-report). They are the relevant team that handles bug related issues and would be able to better assist you.

Hope this helps and clarifies.

**Reply by melvincwng** [reply]
@porg, if this answer has been helpful, kindly 'Mark as answer' to close the thread.

Take care, and have a good day ahead 😄

**Reply by porg** [reply]
@melvincwng wrote:
> Just FYI, in order to get the markdown source code, an alternative way is to simply just copy the original table from the post. Basically just do a Cltrl C and Cltrl V. That works as well.

- That is not true. At least in Safari 15.5 on macOS 11.6.6 Big Sur. I copied the table in output mode, which just copied the pure HTML, and pasting it into a new issue just pasted the HTML. No magic HTML→MarkDown conversion happened. 
- You can only get the Markdown via edit or quoting.
  - Editing someone elses issues is forbidden for good reasons.
  - Quoting is buggy.
  - "Copy source code" would be a new feature (Markdown without the prefixed `>` quote syntax)

**Comment by huabin** [comment]
@melvincwng 

Could you stop begging 'Mark as answer'? In many replies, your reply is always to tell the op to ask GitHub Support for help, and then you beg the op for the `Mark As Answer`.

As I told you several days ago, it's great to want to help others, but if you can't provide a useful and valuable answer, please don't reply, don't tell op to ask GitHub support for help, and DON'T BEG `MARK AS ANSWER`. 

You have too many such replies!!!

**Reply by melvincwng** [reply]
@huabin, I believe I have provided a helpful suggestion as required by OP over here:

```
Just FYI, in order to get the markdown source code, an alternative way is to simply just copy the original table from the post. Basically just do a Cltrl C and Cltrl V. That works as well.
```
**That above is a useful and valuable answer.**

Also, to clarify, OP posted a valid concern that when quoting a reply, it should generate markdown code instead. This I fully concur, hence the suggestion to report it to GitHub Bug Report team for them to investigate (if it's a potential bug). This is not the same as the generic GitHub Support team, based on my understanding. Bugs should be reported to the relevant teams, so that they can be addressed promptly, so that the quality of GitHub's products continuously remain.

You might be surprised, but many GitHub users are not aware of the various support options available to them. I am merely just providing a suggestion to them, that should they wish, they can raise a ticket to the relevant teams. **Instead of their responses being 'ghosted' over here, at least, they will get the support they need.**

With regards to the 'marking of answer', I need to clarify the main purpose is so that others are aware that this issue is addressed/resolved - period. There is **no incentive** for me at all, whether OP wants to mark as answer or not - just to add a disclaimer here.

@liligalante, I understand you are a community manager for GitHub Discussion forums. Please kindly ignore my tag if you are not the right person to address this issue. But, could you kindly lock this discussion thread to prevent further discussions on this topic.

Am just doing my part to help out the community, that's all.

Moving forward, I will still continue to be active in the forums, and (hopefully) provide valuable suggestions and tips to improve or resolve some of the issues which the community might be facing. 

As much as I try to help, one of the main reasons for me being on this forum, is such that not only do I get to learn huge treasure troves of knowledge from answered discussions over here, and I also get to build a network of developers to improve the GitHub experience (or even potentially collaborate on future projects).

Thanks.

**Reply by melvincwng** [reply]
@liligalante, kindly seek your assistance (if possible) on this matter. Thank you, appreciate it.

**Reply by huabin** [reply]
If you agree with what you are doing, you continue.

You can search all your replies and do the math.

How many replies like just telling OP to ask GitHub for help and begging OP to make your reply as answer?

**Reply by melvincwng** [reply]
> You might be surprised, but many GitHub users are not aware of the various support options available to them. I am merely just providing a suggestion to them, that should they wish, they can raise a ticket to the relevant teams. **Instead of their responses being 'ghosted' over here, at least, they will get the support they need.**
> 

>With regards to the 'marking of answer', I need to clarify the main purpose is so that others are aware that this issue is addressed/resolved - period. There is no incentive for me at all, whether OP wants to mark as answer or not - just to add a disclaimer here.
>

As mentioned above, thanks.

**Reply by huabin** [reply]
BTW, I've checked most of your replies, and a lot of the op's questions aren't part of what GitHub support needs to answer, but rather the op should go to google, stackoverflow, etc. There are ready-made answers.

Like I first replied to you a few days ago, the op's question had nothing to do with GitHub at all, it was a question like how to code, but your reply was to tell him to ask GitHub support for help and asked OP to mark your reply as an answer.

ps. you modified your reply after I replied you.

**Reply by melvincwng** [reply]
P.S My original reply here was modified for clarity purposes. Hope this clarifies. Thanks.

**Reply by huabin** [reply]
Not this post, the post where I first asked you not to reply to op's question like this.

On that post, you even asked the OP to follow you on GitHub, and you modified after I asked you not to do that.

**Reply by porg** [reply]
@huabin 
- Thanks for jumping in, and informing about the behavior of @melvincwng
- Melvin's answer was superficial and not helpful.
- I cannot believe that copying a rendered HTML snippet (especially tables, which are complex) and pasting it into a new issue/comment magically creates clean Markdown again in any of the current web browsers. If so, what web browser has that capabilities? Or does the issue submission form on paste analyze the HTML and does a client or server side transformation to Markdown? This seems very fairy tale to me. Do you concur?

**Reply by braxvan** [reply]
I'm back here in 2024 and this issue still has not been solved BTW.  I got directed to this post while googling how to get the markdown out of an issue.

**Comment by porg** [comment]
## Another frustrating use case
- You put quite some text/formatting effort into an issue.
- **An issue get‘s closed. Possibly prematurely. Possible simply for formal reasons** as in „Please open another/new issue in project X again.
- You have to start all over with your formatting efforts.

## I stress again
- The functions „Edit an issue or comment“ and „Copy markup“ should be decoupled/separated! They are not necessarily connected!
- For closed/locked issues/comments offer a „Copy markup“ button.
